src/VkLayer_GF_amber_scoop/src/amber_scoop_layer.cc

```diff
@@ -122,21 +157,245 @@ void init_settings_if_needed() {
     get_setting_string("VkLayer_GF_amber_scoop_OUTPUT_FILE_PREFIX",
                        "debug.gf.as.output_file_prefix",
                        &settings.output_file_prefix);
+    settings.last_draw_call =
+        settings.start_draw_call + settings.draw_call_count;
     settings.init = true;
   }
 }
 
+//
+// Adds the given vulkan command buffer command to the list of tracked commands.
+// Creates a new list of commands for the given command buffer if it doesn't
+// already exist. Tracked commands will be parsed later when the command buffer
+// is submitted via vkQueueSubmit function.
+//
+void AddCommand(const DeviceData& device_data, VkCommandBuffer command_buffer,
```

Comment:
Actually, we might be able to use a heuristic here; track when the command buffer has been submitted. If the app starts recording to such a command buffer then it must have been reset (otherwise I think the app must be violating the spec) so clear our vector before adding the new command.

Reply:
Right, that's a good point!